monkey/monkey_island/cc/environment/environment_config.py

```diff
@@ -35,31 +33,29 @@ def _load_from_file(self, file_path):
 
     def _load_from_json(self, config_json: str) -> EnvironmentConfig:
         data = json.loads(config_json)
-        self._load_from_dict(data)
+        self._load_from_dict(data["environment"])
 
     def _load_from_dict(self, dict_data: Dict):
         aws = dict_data["aws"] if "aws" in dict_data else None
-        data_dir = dict_data["data_dir"] if "data_dir" in dict_data else DEFAULT_DATA_DIR
 
         self.server_config = dict_data["server_config"]
         self.deployment = dict_data["deployment"]
         self.user_creds = _get_user_credentials_from_config(dict_data)
```

Comment:
Why is `_get_user_credentials_from_config()` defined outside `EnvironmentConfig`?

Reply:
It's a utility function that doesn't require any of the state that EnvironmentConfig keeps. Another option would be a private static method, but in python, I think these are not very useful. Python supports first-class functions; use them.

Moreover, user credentials aren't really environment configuration, they're state. So this will go away at some point. See the [above comment](https://github.com/guardicore/monkey/pull/1161#discussion_r631086608).